--- Model/Indexer/Product/Price/AbstractAction.php
+++ Model/Indexer/Product/Price/AbstractAction.php
@@ -177,7 +177,8 @@
         // for backward compatibility split data from old idx table on dimension tables
         foreach ($this->dimensionCollectionFactory->create() as $dimensions) {
             $insertSelect = $this->getConnection()->select()->from(
-                ['ip_tmp' => $this->_defaultIndexerResource->getIdxTable()]
+                ['ip_tmp' => $this->_defaultIndexerResource->getIdxTable()],
+                array_keys($this->getConnection()->describeTable($this->tableMaintainer->getMainTableByDimensions($dimensions)))
             );

             foreach ($dimensions as $dimension) {
