--- Model/ResourceModel/Product/Indexer/Price/Configurable.php.origin	2018-06-19 10:44:22.416411278 +0300
+++ Model/ResourceModel/Product/Indexer/Price/Configurable.php	2018-06-19 10:42:33.369113081 +0300
@@ -176,8 +176,15 @@
         );
         $select->columns(
             [
-                'min_price' => new \Zend_Db_Expr('i.min_price - i.orig_price + io.min_price'),
-                'max_price' => new \Zend_Db_Expr('i.max_price - i.orig_price + io.max_price'),
+                /**
+                 * Applying fix for https://github.com/magento/magento2/issues/15654
+                 * 
+                 * Old:
+                 * 'min_price' => new \Zend_Db_Expr('i.min_price - i.orig_price + io.min_price'),
+                 * 'max_price' => new \Zend_Db_Expr('i.max_price - i.orig_price + io.max_price'),
+                 */
+                'min_price' => new \Zend_Db_Expr('io.min_price'),
+                'max_price' => new \Zend_Db_Expr('io.max_price'),
                 'tier_price' => 'io.tier_price',
             ]
         );
