--- view/frontend/templates/email/shipment/track.orig.phtml	2019-09-19 12:30:56.000000000 +0300
+++ view/frontend/templates/email/shipment/track.phtml	2019-12-06 18:24:23.721943986 +0300
@@ -24,11 +24,7 @@
             <?php foreach ($trackCollection as $_item) : ?>
                 <tr>
                     <td><?= $block->escapeHtml($_item->getTitle()) ?>:</td>
-                    <td>
-                        <a href="<?= $block->escapeUrl($block->getTrackingUrl()->getUrl($_item)) ?>" target="_blank">
-                            <?= $block->escapeHtml($_item->getNumber()) ?>
-                        </a>
-                    </td>
+                    <td><?= $block->escapeHtml($_item->getNumber()) ?></td>
                 </tr>
             <?php endforeach ?>
             </tbody>
