--- Model/Indexer/Product/Flat/Action/Indexer.php
+++ Model/Indexer/Product/Flat/Action/Indexer.php
@@ -96,6 +96,16 @@ class Indexer
                         if (isset($describe[$columnName])) {
                             $ids[$attribute->getId()] = $columnName;
                         }
+                        if (in_array($columnName, [
+                            'special_price',
+                            'cost',
+                            'news_from_date',
+                            'news_to_date',
+                            'special_from_date',
+                            'special_to_date'
+                        ])) {
+                            $updateData[$columnName] = null;
+                        }
                     }
                     $linkField = $this->getMetadataPool()->getMetadata(ProductInterface::class)->getLinkField();
                     $select->joinLeft(
@@ -123,7 +133,7 @@ class Indexer
                     while ($row = $cursor->fetch(\Zend_Db::FETCH_ASSOC)) {
                         $updateData[$ids[$row['attribute_id']]] = $row['value'];
                         $valueColumnName = $ids[$row['attribute_id']] . $valueFieldSuffix;
-                        if (isset($describe[$valueColumnName])) {
+                        if (isset($describe[$valueColumnName]) && (is_int($row['value']) || ctype_digit($row['value']))) {
                             $valueColumns[$row['value']] = $valueColumnName;
                         }
                     }
