--- Controller/Index/Fromcart.php
+++ Controller/Index/Fromcart.php
@@ -112,8 +112,11 @@ class Fromcart extends \Magento\Wishlist\Controller\AbstractIndex
             $buyRequest = $item->getBuyRequest();
             $wishlist->addNewItem($productId, $buyRequest);
 
-            $this->cart->getQuote()->removeItem($itemId);
-            $this->cart->save();
+            $keepItemInCart = (bool)$this->getRequest()->getParam('keep_item_in_cart');
+            if (!$keepItemInCart) {
+                $this->cart->getQuote()->removeItem($itemId);
+                $this->cart->save();
+            }
 
             $this->wishlistHelper->calculate();
             $wishlist->save();
--- Model/ItemCarrier.php
+++ Model/ItemCarrier.php
@@ -147,7 +147,7 @@ class ItemCarrier
                 }
 
                 $cartItem = $cart->getQuote()->getItemByProduct($item->getProduct());
-                if ($cartItem) {
+                if ($cartItem && !$wishlist->getData('keep_item_in_cart')) {
                     $cart->getQuote()->deleteItem($cartItem);
                 }
             } catch (\Exception $e) {
