--- Model/Order/CreditmemoFactory.php
+++ Model/Order/CreditmemoFactory.php
@@ -199,8 +199,12 @@
     protected function initData($creditmemo, $data)
     {
         if (isset($data['shipping_amount'])) {
-            $creditmemo->setBaseShippingAmount((double)$data['shipping_amount']);
-            $creditmemo->setBaseShippingInclTax((double)$data['shipping_amount']);
+            if ($this->taxConfig->displaySalesShippingExclTax() || $this->taxConfig->displaySalesShippingBoth()) {
+                $creditmemo->setBaseShippingAmount((double)$data['shipping_amount']);
+            }
+            if ($this->taxConfig->displaySalesShippingInclTax() || $this->taxConfig->displaySalesShippingBoth()) {
+                $creditmemo->setBaseShippingInclTax((double)$data['shipping_amount']);
+            }
         }
         if (isset($data['adjustment_positive'])) {
             $creditmemo->setAdjustmentPositive($data['adjustment_positive']);
