--- Model/PurgeCache.php
+++ Model/PurgeCache.php
@@ -58,6 +58,7 @@
         $headers = [self::HEADER_X_MAGENTO_TAGS_PATTERN => $tagsPattern];
         $socketAdapter->setOptions(['timeout' => 10]);
         foreach ($servers as $server) {
+            $headers['Host'] = $server->getHost();
             try {
                 $socketAdapter->connect($server->getHost(), $server->getPort());
                 $socketAdapter->write(
@@ -66,6 +67,7 @@
                     '1.1',
                     $headers
                 );
+                $socketAdapter->read();
                 $socketAdapter->close();
             } catch (\Exception $e) {
                 $this->logger->critical($e->getMessage(), compact('server', 'tagsPattern'));
