--- Plugin/Search/Request/ConfigReader.php	2021-01-07 21:56:58.000000000 +0300
+++ Plugin/Search/Request/ConfigReader.php	2021-03-30 08:19:55.000000000 +0300
@@ -106,6 +106,9 @@ class ConfigReader
         $productAttributes->addFieldToFilter(
             ['is_searchable', 'is_visible_in_advanced_search', 'is_filterable', 'is_filterable_in_search'],
             [1, 1, [1, 2], 1]
+        )->setOrder(
+            'position',
+            'ASC'
         );
 
         /** @var Attribute $attribute */
