--- Translate.php	2018-09-04 16:50:34.000000000 +0300
+++ Translate.php	2018-10-03 12:18:45.300379080 +0300
@@ -306,13 +306,14 @@
      * Adding translation data
      *
      * @param array $data
+     * @param bool $force
      * @return $this
      */
-    protected function _addData($data)
+    protected function _addData($data, $force = false)
     {
         foreach ($data as $key => $value) {
             if ($key === $value) {
-                if (isset($this->_data[$key])) {
+                if ($force && isset($this->_data[$key])) {
                     unset($this->_data[$key]);
                 }
                 continue;
@@ -335,7 +336,7 @@
     {
         $file = $this->_getThemeTranslationFile($this->getLocale());
         if ($file) {
-            $this->_addData($this->_getFileData($file));
+            $this->_addData($this->_getFileData($file), true);
         }
         return $this;
     }
@@ -359,7 +360,7 @@
     protected function _loadDbTranslation()
     {
         $data = $this->_translateResource->getTranslationArray(null, $this->getLocale());
-        $this->_addData(array_map('htmlspecialchars_decode', $data));
+        $this->_addData(array_map('htmlspecialchars_decode', $data), true);
         return $this;
     }

