--- Helper/Entity/Product/PriceManager.php	2018-02-27 12:56:18.000000000 +0300
+++ Helper/Entity/Product/PriceManager.php	2018-05-02 20:31:34.000000000 +0300
@@ -62,7 +62,7 @@
             foreach ($currencies as $currencyCode) {
                 $customData[$field][$currencyCode] = [];
 
-                $price = $product->getPrice();
+                $price = $product->getPriceInfo()->getPrice('regular_price')->getValue();
                 if ($currencyCode !== $baseCurrencyCode) {
                     $price = $this->priceCurrency->convert($price, $store, $currencyCode);
                 }