--- HTTP/Client/Curl.php
+++ HTTP/Client/Curl.php
@@ -327,9 +327,13 @@

     /**
      * Make request
+     *
+     * String type was added to parameter $param in order to support sending JSON or XML requests.
+     * This feature was added base on Community Pull Request https://github.com/magento/magento2/pull/8373
+     *
      * @param string $method
      * @param string $uri
-     * @param array $params
+     * @param array|string $params - use $params as a string in case of JSON or XML POST request.
      * @return void
      * @SuppressWarnings(PHPMD.CyclomaticComplexity)
      * @SuppressWarnings(PHPMD.NPathComplexity)
@@ -340,7 +344,7 @@
         $this->curlOption(CURLOPT_URL, $uri);
         if ($method == 'POST') {
             $this->curlOption(CURLOPT_POST, 1);
-            $this->curlOption(CURLOPT_POSTFIELDS, http_build_query($params));
+            $this->curlOption(CURLOPT_POSTFIELDS, is_array($params) ? http_build_query($params) : $params);
         } elseif ($method == "GET") {
             $this->curlOption(CURLOPT_HTTPGET, 1);
         } else {
