--- Model/Resolver/ChangePassword.php
+++ Model/Resolver/ChangePassword.php
@@ -77,16 +77,11 @@
             throw new GraphQlAuthorizationException(__('The current customer isn\'t authorized.'));
         }
 
-        if (!isset($args['currentPassword']) || '' == trim($args['currentPassword'])) {
-            throw new GraphQlInputException(__('Specify the "currentPassword" value.'));
-        }
-
         if (!isset($args['newPassword']) || '' == trim($args['newPassword'])) {
             throw new GraphQlInputException(__('Specify the "newPassword" value.'));
         }
 
         $customerId = $context->getUserId();
-        $this->checkCustomerPassword->execute($args['currentPassword'], $customerId);
 
         try {
             $this->accountManagement->changePasswordById($customerId, $args['currentPassword'], $args['newPassword']);
