diff --git a/lib/web/fotorama/fotorama.js b/lib/web/fotorama/fotorama.js
index a60999ef76474..17736feef448f 100644
--- a/lib/web/fotorama/fotorama.js
+++ b/lib/web/fotorama/fotorama.js
@@ -1140,7 +1140,7 @@ fotoramaVersion = '4.6.4';
 
     function addEvent(el, e, fn, bool) {
         if (!e) return;
-        el.addEventListener ? el.addEventListener(e, fn, !!bool) : el.attachEvent('on' + e, fn);
+        el.addEventListener ? el.addEventListener(e, fn, {passive: !!bool}) : el.attachEvent('on' + e, fn);
     }
 
     /**
