--- Model/Queue.php
+++ Model/Queue.php
@@ -401,7 +401,12 @@

             if ($jobsCount > 0 && $jobsCount < $maxJobs) {
                 $restLimit = $maxJobs - $jobsCount;
-                $lastFullReindexJobId = max($this->getJobsIdsFromMergedJobs($jobs));
+
+                if ($fullReindexJobsCount > 0) {
+                    $lastFullReindexJobId = max($this->getJobsIdsFromMergedJobs($fullReindexJobs));
+                } else {
+                    $lastFullReindexJobId = max($this->getJobsIdsFromMergedJobs($jobs));
+                }

                 $restFullReindexJobs = $this->fetchJobs($restLimit, true, $lastFullReindexJobId);

