--- Model/Order/Creditmemo/Total/Tax.php	2017-01-24 19:38:48.000000000 +0300
+++ Model/Order/Creditmemo/Total/Tax.php	2017-11-20 17:27:55.229021737 +0300
@@ -156,7 +156,12 @@
             $creditmemo->getBaseShippingDiscountTaxCompensationAmnt() -
             $creditmemo->getBaseDiscountTaxCompensationAmount();
 
-        if ($creditmemo->isLast() && !$isPartialShippingRefunded) {
+        if ($invoice) {
+            $totalTax = $invoice->getTaxAmount();
+            $baseTotalTax = $invoice->getBaseTaxAmount();
+            $totalDiscountTaxCompensation = $invoice->getDiscountTaxCompensationAmount();
+            $baseTotalDiscountTaxCompensation = $invoice->getBaseDiscountTaxCompensationAmount();
+        } elseif ($creditmemo->isLast() && !$isPartialShippingRefunded) {
             $totalTax = $allowedTax;
             $baseTotalTax = $allowedBaseTax;
             $totalDiscountTaxCompensation = $allowedDiscountTaxCompensation;