--- Block/Block.php
+++ Block/Block.php
@@ -84,4 +84,11 @@ class Block extends AbstractBlock implements \Magento\Framework\DataObject\Ident
     {
         return [\Magento\Cms\Model\Block::CACHE_TAG . '_' . $this->getBlockId()];
     }
+
+    public function getCacheKeyInfo()
+    {
+        $cacheKeyInfo = parent::getCacheKeyInfo();
+        $cacheKeyInfo[] = $this->_storeManager->getStore()->getId();
+        return $cacheKeyInfo;
+    }
 }