--- Controller/Adminhtml/Category/Save.php.org	2016-08-03 21:35:33.772109252 +0200
+++ Controller/Adminhtml/Category/Save.php	2016-08-03 21:36:33.982229407 +0200
@@ -6,6 +6,7 @@
 namespace Magento\Catalog\Controller\Adminhtml\Category;

 use Magento\Store\Model\StoreManagerInterface;
+use \Magento\Catalog\Api\Data\CategoryAttributeInterface;

 /**
  * Class Save
@@ -49,6 +50,11 @@
     private $storeManager;

     /**
+     * @var \Magento\Eav\Model\Config
+     */
+    private $eavConfig;
+
+    /**
      * Constructor
      *
      * @param \Magento\Backend\App\Action\Context $context
@@ -62,37 +68,15 @@
         \Magento\Framework\Controller\Result\RawFactory $resultRawFactory,
         \Magento\Framework\Controller\Result\JsonFactory $resultJsonFactory,
         \Magento\Framework\View\LayoutFactory $layoutFactory,
-        StoreManagerInterface $storeManager
+        StoreManagerInterface $storeManager,
+        \Magento\Eav\Model\Config $eavConfig
     ) {
         parent::__construct($context);
         $this->resultRawFactory = $resultRawFactory;
         $this->resultJsonFactory = $resultJsonFactory;
         $this->layoutFactory = $layoutFactory;
         $this->storeManager = $storeManager;
-    }
-
-    /**
-     * Filter category data
-     *
-     * @param array $rawData
-     * @return array
-     */
-    protected function _filterCategoryPostData(array $rawData)
-    {
-        $data = $rawData;
-        // @todo It is a workaround to prevent saving this data in category model and it has to be refactored in future
-        if (isset($data['image']) && is_array($data['image'])) {
-            if (!empty($data['image']['delete'])) {
-                $data['image'] = null;
-            } else {
-                if (isset($data['image'][0]['name']) && isset($data['image'][0]['tmp_name'])) {
-                    $data['image'] = $data['image'][0]['name'];
-                } else {
-                    unset($data['image']);
-                }
-            }
-        }
-        return $data;
+        $this->eavConfig = $eavConfig;
     }

     /**
@@ -126,7 +110,7 @@
         $this->storeManager->setCurrentStore($store->getCode());
         $parentId = isset($categoryPostData['parent']) ? $categoryPostData['parent'] : null;
         if ($categoryPostData) {
-            $category->addData($this->_filterCategoryPostData($categoryPostData));
+            $category->addData($categoryPostData);
             if ($isNewCategory) {
                 $parentCategory = $this->getParentCategory($parentId, $storeId);
                 $category->setPath($parentCategory->getPath());
@@ -247,22 +231,29 @@
         );
     }

-    /**
-     * Image data preprocessing
-     *
-     * @param array $data
-     *
-     * @return array
-     */
     public function imagePreprocessing($data)
     {
-        if (empty($data['image'])) {
-            unset($data['image']);
-            $data['image']['delete'] = true;
+        $entityType = $this->eavConfig->getEntityType(CategoryAttributeInterface::ENTITY_TYPE_CODE);
+
+        foreach ($entityType->getAttributeCollection() as $attributeModel) {
+            $attributeCode = $attributeModel->getAttributeCode();
+            $backendModel = $attributeModel->getBackend();
+
+            if (isset($data[$attributeCode])) {
+                continue;
+            }
+
+            if (!$backendModel instanceof \Magento\Catalog\Model\Category\Attribute\Backend\Image) {
+                continue;
+            }
+
+            $data[$attributeCode] = false;
         }
+
         return $data;
     }

+
     /**
      * Converting inputs from string to boolean
      *
--- Controller/Adminhtml/Category/Image/Upload.php.org	2016-08-01 20:36:22.014237780 +0200
+++ Controller/Adminhtml/Category/Image/Upload.php	2016-08-01 20:36:25.292475257 +0200
@@ -50,8 +50,10 @@
      */
     public function execute()
     {
+        $imageId = $this->_request->getParam('param_name', 'image');
+
         try {
-            $result = $this->imageUploader->saveFileToTmpDir('image');
+            $result = $this->imageUploader->saveFileToTmpDir($imageId);

             $result['cookie'] = [
                 'name' => $this->_getSession()->getName(),
--- Model/Category/Attribute/Backend/Image.php.org	2016-08-03 21:36:05.695112561 +0200
+++ Model/Category/Attribute/Backend/Image.php	2016-08-03 21:36:46.829828298 +0200
@@ -13,6 +13,8 @@

 class Image extends \Magento\Eav\Model\Entity\Attribute\Backend\AbstractBackend
 {
+    const ADDITIONAL_DATA_SUFFIX = '_additional_data';
+
     /**
      * @var \Magento\MediaStorage\Model\File\UploaderFactory
      *
@@ -21,8 +23,6 @@
     protected $_uploaderFactory;

     /**
-     * Filesystem facade
-     *
      * @var \Magento\Framework\Filesystem
      *
      * @deprecated
@@ -30,8 +30,6 @@
     protected $_filesystem;

     /**
-     * File Uploader factory
-     *
      * @var \Magento\MediaStorage\Model\File\UploaderFactory
      *
      * @deprecated
@@ -46,15 +44,11 @@
     protected $_logger;

     /**
-     * Image uploader
-     *
      * @var \Magento\Catalog\Model\ImageUploader
      */
     private $imageUploader;

     /**
-     * Image constructor.
-     *
      * @param \Psr\Log\LoggerInterface $logger
      * @param \Magento\Framework\Filesystem $filesystem
      * @param \Magento\MediaStorage\Model\File\UploaderFactory $fileUploaderFactory
@@ -70,8 +64,50 @@
     }

     /**
-     * Get image uploader
+     * @param $value
+     * @return string|bool
+     */
+    protected function getUploadedImageName($value)
+    {
+        if (!is_array($value)) {
+            return false;
+        }
+
+        if (!count($value)) {
+            return false;
+        }
+
+        $imageData = reset($value);
+
+        if (!isset($imageData['name'])) {
+            return false;
+        }
+
+        return $imageData['name'];
+    }
+
+    /**
+     * Avoiding saving potential upload data to DB
      *
+     * @param \Magento\Framework\DataObject $object
+     * @return $this
+     */
+    public function beforeSave($object)
+    {
+        $attributeName = $this->getAttribute()->getName();
+        $value = $object->getData($attributeName);
+
+        if ($value === false || (is_array($value) && isset($value['delete']) && $value['delete'] === true)) {
+            $object->setData($attributeName, '');
+        } else if ($imageName = $this->getUploadedImageName($value)) {
+            $object->setData($attributeName . self::ADDITIONAL_DATA_SUFFIX, $value);
+            $object->setData($attributeName, $imageName);
+        }
+
+        return parent::beforeSave($object);
+    }
+
+    /**
      * @return \Magento\Catalog\Model\ImageUploader
      *
      * @deprecated
@@ -80,9 +116,10 @@
     {
         if ($this->imageUploader === null) {
             $this->imageUploader = \Magento\Framework\App\ObjectManager::getInstance()->get(
-                'Magento\Catalog\CategoryImageUpload'
+                \Magento\Catalog\CategoryImageUpload::class
             );
         }
+
         return $this->imageUploader;
     }

@@ -94,15 +131,18 @@
      */
     public function afterSave($object)
     {
-        $image = $object->getData($this->getAttribute()->getName(), null);
+        $value = $object->getData($this->getAttribute()->getName() . self::ADDITIONAL_DATA_SUFFIX);

-        if ($image !== null) {
-            try {
-                $this->getImageUploader()->moveFileFromTmp($image);
-            } catch (\Exception $e) {
-                $this->_logger->critical($e);
-            }
+        if (!$imageName = $this->getUploadedImageName($value)) {
+            return $this;
         }
+
+        try {
+            $this->getImageUploader()->moveFileFromTmp($imageName);
+        } catch (\Exception $e) {
+            $this->_logger->critical($e);
+        }
+
         return $this;
     }
 }
--- Model/Category/DataProvider.php.org	2016-08-01 21:35:43.567609510 +0200
+++ Model/Category/DataProvider.php	2016-08-01 21:43:07.800993338 +0200
@@ -203,14 +203,24 @@
         $category = $this->getCurrentCategory();
         if ($category) {
             $categoryData = $category->getData();
+
             $categoryData = $this->addUseDefaultSettings($category, $categoryData);
             $categoryData = $this->addUseConfigSettings($categoryData);
             $categoryData = $this->filterFields($categoryData);
-            if (isset($categoryData['image'])) {
-                unset($categoryData['image']);
-                $categoryData['image'][0]['name'] = $category->getData('image');
-                $categoryData['image'][0]['url'] = $category->getImageUrl();
+
+            foreach ($category->getAttributes() as $attributeCode => $attribute) {
+                $backendModel = $attribute->getBackend();
+
+                if ($backendModel instanceof \Magento\Catalog\Model\Category\Attribute\Backend\Image) {
+                    if (isset($categoryData[$attributeCode])) {
+                        unset($categoryData[$attributeCode]);
+
+                        $categoryData[$attributeCode][0]['name'] = $category->getData($attributeCode);
+                        $categoryData[$attributeCode][0]['url'] = $category->getImageUrl($attributeCode);
+                    }
+                }
             }
+
             $this->loadedData[$category->getId()] = $categoryData;
         }
         return $this->loadedData;
--- Model/Category.php.org	2016-08-01 21:41:47.535876208 +0200
+++ Model/Category.php	2016-08-01 21:42:57.916422400 +0200
@@ -652,14 +652,15 @@
     }

     /**
-     * Retrieve image URL
+     * @param $attributeCode
      *
-     * @return string
+     * @return bool|string
+     * @throws \Magento\Framework\Exception\LocalizedException
      */
-    public function getImageUrl()
+    public function getImageUrl($attributeCode = 'image')
     {
         $url = false;
-        $image = $this->getImage();
+        $image = $this->getData($attributeCode);
         if ($image) {
             if (is_string($image)) {
                 $url = $this->_storeManager->getStore()->getBaseUrl(