--- Model/ResourceModel/Indexer/Stock/DefaultStock.php
+++ Model/ResourceModel/Indexer/Stock/DefaultStock.php
@@ -292,6 +292,7 @@
      */
     protected function _updateIndex($entityIds)
     {
+        $this->deleteOldRecords($entityIds);
         $connection = $this->getConnection();
         $select = $this->_getStockStatusSelect($entityIds, true);
         $select = $this->getQueryProcessorComposite()->processQuery($select, $entityIds, true);
@@ -314,7 +315,6 @@
             }
         }
 
-        $this->deleteOldRecords($entityIds);
         $this->_updateIndexTable($data);
 
         return $this;
