--- Model/ResourceModel/Category.php
+++ Model/ResourceModel/Category.php
@@ -415,6 +415,7 @@ class Category extends AbstractResource
                 'catalog_category_change_products',
                 ['category' => $category, 'product_ids' => $productIds]
             );
+            $category->setUrlRegenerationRequired(true);
         }
 
         if (!empty($insert) || !empty($update) || !empty($delete)) {
