--- Mail/Message.orig.php	2019-09-19 12:30:58.000000000 +0300
+++ Mail/Message.php	2019-11-27 11:03:14.055398987 +0300
@@ -168,7 +168,6 @@
         $part = new Part($body);
         $part->setCharset($this->zendMessage->getEncoding());
         $part->setEncoding(Mime::ENCODING_QUOTEDPRINTABLE);
-        $part->setDisposition(Mime::DISPOSITION_INLINE);
         $part->setType($messageType);
         $mimeMessage = new \Zend\Mime\Message();
         $mimeMessage->addPart($part);
--- Mail/MimePart.orig.php	2019-09-19 12:30:58.000000000 +0300
+++ Mail/MimePart.php	2019-11-27 11:03:52.275935922 +0300
@@ -47,7 +47,7 @@
         $content,
         ?string $type = MimeInterface::TYPE_HTML,
         ?string $fileName = null,
-        ?string $disposition = MimeInterface::DISPOSITION_INLINE,
+        ?string $disposition = null,
         ?string $encoding = MimeInterface::ENCODING_QUOTED_PRINTABLE,
         ?string $description = null,
         ?array $filters = [],
