--- Model/Context/AddCustomerGroupToContext.php	1970-01-01 03:00:00.000000000 +0300
+++ Model/Context/AddCustomerGroupToContext.php	2020-12-23 00:15:09.000000000 +0300
@@ -0,0 +1,52 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CustomerGraphQl\Model\Context;
+
+use Magento\Authorization\Model\UserContextInterface;
+use Magento\GraphQl\Model\Query\ContextParametersInterface;
+use Magento\GraphQl\Model\Query\ContextParametersProcessorInterface;
+use Magento\Customer\Model\Group;
+use Magento\Customer\Model\Group\Resolver as CustomerGroupResolver;
+
+/**
+ * @inheritdoc
+ */
+class AddCustomerGroupToContext implements ContextParametersProcessorInterface
+{
+    /**
+     * @var CustomerGroupResolver
+     */
+    private $customerGroupResolver;
+
+    /**
+     * @param CustomerGroupResolver $customerGroupResolver
+     */
+    public function __construct(
+        CustomerGroupResolver $customerGroupResolver
+    ) {
+        $this->customerGroupResolver = $customerGroupResolver;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function execute(ContextParametersInterface $contextParameters): ContextParametersInterface
+    {
+        $customerGroupId = null;
+        $extensionAttributes = $contextParameters->getExtensionAttributesData();
+        if ($contextParameters->getUserType() === UserContextInterface::USER_TYPE_GUEST) {
+            $customerGroupId = Group::NOT_LOGGED_IN_ID;
+        } elseif (!empty($extensionAttributes) && $extensionAttributes['is_customer'] === true) {
+            $customerGroupId = $this->customerGroupResolver->resolve((int) $contextParameters->getUserId());
+        }
+        if ($customerGroupId !== null) {
+            $contextParameters->addExtensionAttribute('customer_group_id', (int) $customerGroupId);
+        }
+        return $contextParameters;
+    }
+}
\ No newline at end of file
--- etc/graphql/di.xml	2020-09-23 20:00:16.000000000 +0300
+++ etc/graphql/di.xml	2020-12-23 00:18:24.000000000 +0300
@@ -17,6 +17,7 @@
         <arguments>
             <argument name="contextParametersProcessors" xsi:type="array">
                 <item name="add_user_info_to_context" xsi:type="object">Magento\CustomerGraphQl\Model\Context\AddUserInfoToContext</item>
+                <item name="add_customer_group_to_context" xsi:type="object">Magento\CustomerGraphQl\Model\Context\AddCustomerGroupToContext</item>
             </argument>
         </arguments>
     </type>
