--- view/frontend/web/js/action/select-payment-method-mixin.js	2020-04-13 17:35:42.000000000 +0300
+++ view/frontend/web/js/action/select-payment-method-mixin.js	2020-06-19 15:24:44.503492879 +0300
@@ -7,10 +7,10 @@
     'mage/utils/wrapper',
     'Magento_Checkout/js/model/quote',
     'Magento_SalesRule/js/model/payment/discount-messages',
-    'Magento_Checkout/js/action/set-payment-information',
+    'Magento_Checkout/js/action/set-payment-information-extended',
     'Magento_Checkout/js/action/get-totals',
     'Magento_SalesRule/js/model/coupon'
-], function ($, wrapper, quote, messageContainer, setPaymentInformationAction, getTotalsAction, coupon) {
+], function ($, wrapper, quote, messageContainer, setPaymentInformationExtended, getTotalsAction, coupon) {
     'use strict';

     return function (selectPaymentMethodAction) {
@@ -19,12 +19,17 @@

             originalSelectPaymentMethodAction(paymentMethod);

+            if (paymentMethod === null) {
+                return;
+            }
+
             $.when(
-                setPaymentInformationAction(
+                setPaymentInformationExtended(
                     messageContainer,
                     {
                         method: paymentMethod.method
-                    }
+                    },
+                    true
                 )
             ).done(
                 function () {
@@ -47,4 +52,4 @@
         });
     };

-});
+});
\ No newline at end of file
