--- Model/DataProvider/RelatedProductDataProvider.php	2021-01-07 21:56:58.000000000 +0300
+++ Model/DataProvider/RelatedProductDataProvider.php	2021-05-27 15:12:25.000000000 +0300
@@ -118,6 +118,8 @@
         $collection = $link->getLinkCollection();
         $collection->addFieldToFilter('product_id', ['in' => array_keys($productsByActualIds)]);
         $collection->addLinkTypeIdFilter();
+        $collection->joinAttributes();
+        $collection->addOrder('position', 'asc');
 
         //Prepare map
         $map = [];
