--- Controller/Payment/Klarna/InitRegular.php
+++ Controller/Payment/Klarna/InitRegular.php
@@ -218,7 +218,7 @@

         $this->setCheckoutMethod($quote);

-        $maskedQuoteId = $this->quoteIdToMaskedQuoteId->execute($quote->getId());
+        $maskedQuoteId = $this->quoteIdToMaskedQuoteId->execute((int)$quote->getId());
         switch ($quote->getCheckoutMethod()) {
             case Onepage::METHOD_CUSTOMER:
                 $this->paymentInformationManagement->savePaymentInformationAndPlaceOrder(
