--- Model/Indexer/Category/Product/AbstractAction.php   2019-07-09 14:57:01.280954957 +0300
+++ Model/Indexer/Category/Product/AbstractAction.php   2019-07-09 14:56:04.000000000 +0300
@@ -174,9 +174,9 @@
         foreach ($this->storeManager->getStores() as $store) {
             if ($this->getPathFromCategoryId($store->getRootCategoryId())) {
                 $this->currentStoreId = $store->getId();
-                $this->reindexRootCategory($store);
                 $this->reindexAnchorCategories($store);
                 $this->reindexNonAnchorCategories($store);
+                $this->reindexRootCategory($store);
             }
         }
     }
@@ -860,8 +860,11 @@
     protected function reindexRootCategory(Store $store)
     {
         if ($this->isIndexRootCategoryNeeded()) {
+            $productSelect = $this->getAllProducts($store)->where(
+                'ccp.category_id = ' . $store->getRootCategoryId()
+            );
             $selects = $this->prepareSelectsByRange(
-                $this->getAllProducts($store),
+                $productSelect,
                 'entity_id',
                 self::RANGE_PRODUCT_STEP
             );

--- Model/Indexer/Category/Product/Action/Full.php
+++ Model/Indexer/Category/Product/Action/Full.php
@@ -272,4 +272,15 @@
             $this->publishData($store);
         }
     }
+
+    /**
+     * Check whether indexation of root category is needed
+     *
+     * @return bool
+     */
+    protected function isIndexRootCategoryNeeded()
+    {
+        return false;
+    }
 }
+
