--- Model/QuoteLocator.php	2022-10-31 13:57:44.000000000 +0100
+++ Model/QuoteLocator.php	2022-11-03 11:30:42.000000000 +0100
@@ -18,7 +18,7 @@
 use Magento\Quote\Api\Data\CartInterface;
 use Magento\Quote\Model\Quote;
 use Magento\Quote\Model\QuoteRepository;
-use Vipps\Checkout\Api\QuoteRepositoryInterface as VippsQuoteRepositoryInterface;
+use Vipps\Payment\Api\QuoteRepositoryInterface as VippsQuoteRepositoryInterface;

 /**
  * Class QuoteLocator
