--- Plugin/ShippingInformationPlugin.php	2025-03-12 08:29:15.426475610 +0100
+++ Plugin/ShippingInformationPlugin.php	2025-03-12 08:29:07.920275344 +0100
@@ -53,7 +53,7 @@
         $rate
     ): ShippingMethodInterface {
         try {
-            if (strpos($result->getCarrierCode(), 'bring') === false) {
+            if (strpos((string) $result->getCarrierCode(), 'bring') === false) {
                 return $result;
             }
 
