--- Model/Resolver/Batch/AbstractLikedProducts.php	2020-07-20 11:13:10.000000000 +0000
+++ Model/Resolver/Batch/AbstractLikedProducts.php	2021-03-03 13:55:38.897669052 +0000
@@ -110,6 +110,10 @@
         //Matching products with related products.
         $relationsData = [];
         foreach ($relations as $productId => $relatedIds) {
+            //Remove related products that not exist in map list.
+            $relatedIds = array_filter($relatedIds, function ($relatedId) use ($relatedProducts) {
+                return isset($relatedProducts[$relatedId]);
+            });
             $relationsData[$productId] = array_map(
                 function ($id) use ($relatedProducts) {
                     return $relatedProducts[$id];
