--- Model/Plugin/EavAttribute.php
+++ Model/Plugin/EavAttribute.php
@@ -228,19 +228,19 @@ class EavAttribute
     {
         $swatchArray = $attribute->getData('swatch/value');
         if (isset($swatchArray) && is_array($swatchArray)) {
-            foreach ($swatchArray as $optionId => $value) {
+            foreach ($swatchArray as $optionId => $storeValues) {
                 $optionId = $this->getAttributeOptionId($optionId);
                 $isOptionForDelete = $this->isOptionForDelete($attribute, $optionId);
                 if ($optionId === null || $isOptionForDelete) {
                     //option was deleted by button with basket
                     continue;
                 }
-                $swatch = $this->loadSwatchIfExists($optionId, self::DEFAULT_STORE_ID);
-
-                $swatchType = $this->determineSwatchType($value);
-
-                $this->saveSwatchData($swatch, $optionId, self::DEFAULT_STORE_ID, $swatchType, $value);
-                $this->isSwatchExists = null;
+                foreach ($storeValues as $storeId => $value) {
+                    $swatch = $this->loadSwatchIfExists($optionId, $storeId);
+                    $swatchType = $this->determineSwatchType($value);
+                    $this->saveSwatchData($swatch, $optionId, $storeId, $swatchType, $value);
+                    $this->isSwatchExists = null;
+                }
             }
         }
     }
