--- Pricing/Price/LowestPriceOptionsProvider.php
+++ Pricing/Price/LowestPriceOptionsProvider.php
@@ -55,7 +55,9 @@
         );
 
         $lowestPriceChildProducts = $this->collectionFactory->create()
-            ->addAttributeToSelect(['price', 'special_price'])
+            ->addAttributeToSelect(
+                ['price', 'special_price', 'special_from_date', 'special_to_date', 'tax_class_id']
+            )
             ->addIdFilter($productIds)
             ->getItems();
         return $lowestPriceChildProducts;
