Index: Model/Product/Attribute/Backend/TierPrice/UpdateHandler.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Model/Product/Attribute/Backend/TierPrice/UpdateHandler.php	(revision 1804146a601561685f54c920bf869b3f2398cb9e)
+++ Model/Product/Attribute/Backend/TierPrice/UpdateHandler.php	(date 1537531110000)
@@ -83,7 +83,7 @@
                 );
             }
             $websiteId = $this->storeManager->getStore($entity->getStoreId())->getWebsiteId();
-            $isGlobal = $attribute->isScopeGlobal() || $websiteId === 0;
+            $isGlobal = $attribute->isScopeGlobal() || $this->isWebsiteGlobal((int)$websiteId);
             $identifierField = $this->metadataPoll->getMetadata(ProductInterface::class)->getLinkField();
             $productId = $entity->getData($identifierField);
 
Index: Model/Product/Attribute/Backend/TierPrice/SaveHandler.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Model/Product/Attribute/Backend/TierPrice/SaveHandler.php	(revision 1804146a601561685f54c920bf869b3f2398cb9e)
+++ Model/Product/Attribute/Backend/TierPrice/SaveHandler.php	(date 1537531110000)
@@ -86,7 +86,7 @@
                 );
             }
             $websiteId = $this->storeManager->getStore($entity->getStoreId())->getWebsiteId();
-            $isGlobal = $attribute->isScopeGlobal() || $websiteId === 0;
+            $isGlobal = $attribute->isScopeGlobal() || $this->isWebsiteGlobal((int)$websiteId);
             $identifierField = $this->metadataPoll->getMetadata(ProductInterface::class)->getLinkField();
             $priceRows = array_filter($priceRows);
             $productId = $entity->getData($identifierField);
