--- Model/ShippingInformationManagement.orig.php	2019-09-19 12:30:52.000000000 +0300
+++ Model/ShippingInformationManagement.php	2019-11-20 14:46:58.532977852 +0300
@@ -194,9 +194,15 @@
         if (!$quote->getIsVirtual()
             && !$shippingAddress->getShippingRateByCode($shippingAddress->getShippingMethod())
         ) {
-            throw new NoSuchEntityException(
-                __('Carrier with such method not found: %1, %2', $carrierCode, $methodCode)
+            $this->logger->warning(
+                __('Carrier with such method not found: %1, %2', $carrierCode, $methodCode),
+                [
+                    'quote_id' => $cartId,
+                    'carrier_code' => $carrierCode,
+                    'method_code' => $methodCode,
+                ]
             );
+            // don't throw exception
         }
 
         /** @var \Magento\Checkout\Api\Data\PaymentDetailsInterface $paymentDetails */
