--- view/frontend/templates/vconnect/embedded/base.phtml	2018-12-06 19:01:09.690438289 +0300
+++ view/frontend/templates/vconnect/embedded/base.phtml	2018-12-06 19:00:12.097411516 +0300
@@ -157,7 +157,16 @@
 
     <script>
         jQuery(document).ready(function ($) {
-            var onCoordsLoad = new Event('onCoordsLoad');
+            var onCoordsLoad;
+            var eventName = 'onCoordsLoad';
+
+            if (typeof(Event) === 'function') {
+                onCoordsLoad = new Event(eventName);
+            } else {
+                onCoordsLoad = document.createEvent('Event');
+                onCoordsLoad.initEvent(eventName, true, true);
+            }
+
             (function(d, id){
                 if (d.getElementById(id)) {
                     return;
