--- Plugin/TemplatePlugin.php	2018-12-19 19:18:24.749573848 +0300
+++ Plugin/TemplatePlugin.php	2018-12-19 19:36:36.259173869 +0300
@@ -137,7 +137,7 @@
                 }
                 //decode encoded subject
                 if ($field == 'template_subject') {
-                    $result = utf8_decode($result);
+                    $result = "=?utf-8?B?" . base64_encode($result) . "?=";
                 }
                 if ($field == 'template_id') {
                     $this->saveTemplateIdInRegistry($result);
@@ -160,7 +160,6 @@
         }
         if (isset($result['template_text'])) {
             $templateText = $result['template_text'];
-            $result['template_subject'] = utf8_decode($result['template_subject']);
             if ($this->isStringCompressed($templateText)) {
                 $result['template_text'] = $this->decompresString($templateText);
             }
